cmd/lotus-shed/datastore-vlog.go

```diff
@@ -0,0 +1,342 @@
+package main
+
+import (
+	"bufio"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"hash"
+	"hash/crc32"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/dgraph-io/badger/v2/y"
+	block "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-base32"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+)
+
+var datastoreVlog2CarCmd = &cli.Command{
+	Name:  "vlog2car",
+	Usage: "convert badger blockstore .vlog to .car",
```

Comment:
Any warnings needed about the daemon needing to be turned off as in the following utility?

Reply:
I'm not seeing any interaction with the repo lock file, any chance for problems if the node is online?

Reply:
vlog files are append-only, so that's (probably) fine